src/OrchardCore.Modules/OrchardCore.Markdown/Handlers/MarkdownBodyPartHandler.cs

```diff
@@ -35,19 +55,34 @@ public override Task GetContentItemAspectAsync(ContentItemAspectContext context,
 
                 try
                 {
-                    var model = new MarkdownBodyPartViewModel()
+                    var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(part.ContentItem.ContentType);
+                    var contentTypePartDefinition = contentTypeDefinition.Parts.FirstOrDefault(x => String.Equals(x.PartDefinition.Name, "MarkdownBodyPart"));
+                    var settings = contentTypePartDefinition.GetSettings<MarkdownBodyPartSettings>();
+
+                    var markdown = part.Markdown;
+
+                    if (settings.AllowCustomScripts)
                     {
-                        Markdown = part.Markdown,
-                        MarkdownBodyPart = part,
-                        ContentItem = part.ContentItem
-                    };
+                        var model = new MarkdownBodyPartViewModel()
+                        {
+                            Markdown = part.Markdown,
+                            MarkdownBodyPart = part,
+                            ContentItem = part.ContentItem
+                        };
 
-                    var markdown = await _liquidTemplateManager.RenderAsync(part.Markdown, _htmlEncoder, model,
-                        scope => scope.SetValue("ContentItem", model.ContentItem));
+                        markdown = await _liquidTemplateManager.RenderAsync(markdown, _htmlEncoder, model,
+                            scope => scope.SetValue("ContentItem", model.ContentItem));
+                    }
 
-                    var result = Markdig.Markdown.ToHtml(markdown ?? "");
+                    markdown = await _safeCodeFilterManager.ProcessAsync(markdown);
+                    markdown = _markdownService.ToHtml(markdown);
+
+                    if (!settings.AllowCustomScripts)
```

Comment:
change this setting to `settings.Sanitize` to reverse the test, and make code more readable. Default to true